### DISCUSSION TITLE
List all orders by email so guest can find their orders once registered

### DISCUSSION THREAD

**Discussion by winston0410** [medusajs/medusa#4437] [Feature Requests]
Right now I can see there is a route dedicated to find order by `email` and `display_id`: https://docs.medusajs.com/api/store#tag/Orders/operation/GetOrders

With that route, it can only find a specific order. A use case I found useful is, I want guest to check all their historical orders after signed up as a customer, and I want to do that by just searching with `email`. Can we make `display_id` optional, or is there another route for that?

**Comment by srindom** [comment]
The registered-account-to-order flow works as follows:

- A guest customer (i.e. not logged in) with email a@mail.com creates an order (**Order 1**). The associated order.customer_id = "a"
- The same customer registers an account with your store using the same email. However, the newly registered account will have id = "b"
- The customer creates an order (**Order 2**) while logged in. The associated order.customer = "b"
- The customer wants to see their historical orders. These are fetched with `GET /customers/me/orders`, in the result only Order 2 is displayed.

This behavior is intentional as we wouldn't want guest orders to be "hijacked" by customers who create accounts for emails they don't own - the distinct customer ids for registered vs. guest checkouts ensures this.

We have a mechanism for "transferring" a guest order to a registered account. This can either be done through the [admin dashboard](https://docs.medusajs.com/user-guide/orders/manage#transfer-customer-ownership). Alternatively, you can build self-serve into your storefront using the [Claim an Order endpoint](https://docs.medusajs.com/user-guide/orders/manage#transfer-customer-ownership).

The approach would be something like this:
1. In your storefront, allow a customer to look up an order using `email`, `display_id`. 
2. When the order is retrieved and displayed to the customer, create a button: "Transfer to account".
3. When the button is clicked you should call the claim an order endpoint. This will trigger a `TokenEvents.ORDER_UPDATE_TOKEN_CREATED` event that you subscribe to and send an email to the user. [You can see the code here.](https://github.com/medusajs/medusa/blob/develop/packages/medusa/src/api/routes/store/orders/request-order.ts)
4. In the email a link should allow the customer to verify the transfer using the token created as part of the Claim an Order endpoint.

**Reply by winston0410** [reply]
That is really good to know and explained why it is designed this way. Thank you so much and I will try and implement that order claim functionality
